src/App.tsx

```diff
@@ -120,12 +121,9 @@ function App() {
     }
   }
 
-  const initNetwork = useCallback(
-    loadProvider => {
-      dispatch(initializeNetwork({ provider: loadProvider }));
-    },
-    [networkId],
-  );
+  const initNetwork = useCallback(loadProvider => {
+    dispatch(initializeNetwork({ provider: loadProvider }));
+  }, []);
```

Comment:
ah hell. this is a good catch. the callback should definitely NOT be refreshed on `networkId`. I have one question: Shouldn't the callback be refreshed on `provider` since `provider` is what generates the networkId within `initializeNetwork` (see screenshot)?
![image](https://user-images.githubusercontent.com/80423742/144942399-c32afa12-0077-4b68-8967-456e46d6a026.png)

Reply:
actually upon further review we probably don't need `provider` in the dependencies array since we are passing in `loadProvider`

Reply:
> we are passing in loadProvider

Exactly